backend/pkg/config/config.go

```diff
@@ -155,6 +158,46 @@ func flagset() *flag.FlagSet {
 	return f
 }
 
+// defaultKubeConfigPersistenceDir returns the default directory to store kubeconfig
+// files of clusters that are loaded in Headlamp.
+func defaultKubeConfigPersistenceDir() string {
+	// These are the folders we use for the default kubeconfig-dir.
+	//  - the passed in kubeconfig-dir if it's not empty.
+	//  - "./.kubeconfigs" if it exists.
+	//  - ~/.config/Headlamp/kubeconfigs exists or it can be made
+	//  - "./.kubeconfigs" if the ~/.config/Headlamp/kubeconfigs can't be made.
+	// Windows: %APPDATA%\Headlamp\Config\kubeconfigs
+	//   (for example, C:\Users\USERNAME\AppData\Roaming\Headlamp\Config\plugins)
+	kubeConfigsDirDefault := "./.kubeconfigs"
```

Comment:
This is a fallback value to be used if the config folder doesn't exist.

Reply:
But the config folder should really be the default. If it doesn't exist, we shouldn't really look for another path unless it's common knowledge that it should be there.

Reply:
I tried to make the experience similar to how the default plugins folder is selected. The config directory might not exist or be found when Headlamp is running in a container, In that case it will default to ./.kubeconfigs directory. If the config folder exists it will be used.